### PR TITLE
Add coverage report

### PR DESCRIPTION
**Pull request by sheb**
It's quite good but we can do better :)

```
----------- coverage: platform linux, python 3.6.4-final-0 -----------
Name                 Stmts   Miss  Cover   Missing
--------------------------------------------------
west/__init__.py        10      5    50%   7-15
west/__main__.py         3      1    67%   4
west/cli.py             57      6    89%   39, 47-48, 165, 186, 207
west/core.py            74     31    58%   21-25, 39-40, 47-58, 66, 81, 97, 112-113, 138-157
west/db.py              63      1    98%   102
west/exceptions.py       2      0   100%
west/files.py           52      6    88%   30-31, 49, 87-88, 105
west/migrate.py         84     12    86%   47-50, 66-70, 108-109, 131-136
west/runner.py          98      6    94%   34, 39, 48, 76, 88-89
west/settings.py        25      2    92%   23-26
west/style.py           34      0   100%
west/utils.py           15      2    87%   15, 44
--------------------------------------------------
TOTAL                  517     72    86%
```

### CODE REVIEW

**Comment by k4nar** [comment]
Great :) .

Shouldn't we try to use a tool like https://codecov.io/ ? This is rather easy to add, cf this example: https://github.com/peopledoc/populous/blob/master/.travis.yml#L19-L26

**Comment by ewjoachim** [comment]
We said we wanted to separate integration coverage from unit coverage. What of it ?

**Comment by sheb** [comment]
If I understand well it could be done using flags ? : https://docs.codecov.io/docs/flags

**Comment by ewjoachim** [comment]
Their example is exactly our usecase so I say yes ! :D (that's so cool !)

screenshot from 2018-09-06 19-08-32

File: tox.ini [hunk 1]
```diff
@@ -7,9 +7,11 @@ whitelist_externals = make
 usedevelop = True
 extras =
     test
+passenv = PGPASSWORD PGHOST PGUSER PGPORT
 commands =
     /usr/bin/find . -name '*.pyc' -delete
     pip freeze -l
+    pytest --cov-report term-missing --cov=west tests/unittests
```
Comment (k4nar) [inline tox.ini]: so we don't launch the integration tests in the CI?

Comment (sheb) [reply tox.ini]: coming in the next PR :)

**Pull request merged**